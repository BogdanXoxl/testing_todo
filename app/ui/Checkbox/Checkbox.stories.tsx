import type { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";

import { Checkbox } from "./Checkbox";

export default {
  title: "UI/Checkbox",
  component: Checkbox,
  argTypes: {
    checked: {
      type: "boolean",
      defaultValue: false,
    },
    className: {
      type: "string",
      control: false,
    },
  },
} as ComponentMeta<typeof Checkbox>;

const Template: ComponentStory<typeof Checkbox> = (args) => <Checkbox {...args} />;

export const Default = Template.bind({});
