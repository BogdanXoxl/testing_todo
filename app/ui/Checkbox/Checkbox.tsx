import clsx from "clsx";
import React from "react";
import { BsCheck } from "react-icons/bs";

type Props = {
  checked?: boolean;
  className?: string;
};

export const Checkbox: React.FC<Props> = ({ checked = false, className }) => {
  return (
    <div
      className={clsx(
        "h-7 w-7 shrink-0 rounded-lg border-2 border-primary",
        checked && "bg-primary",
        className
      )}
    >
      {checked && <BsCheck size={24} className="text-secondary" />}
    </div>
  );
};
