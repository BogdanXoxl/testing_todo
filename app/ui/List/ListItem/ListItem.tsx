import clsx from "clsx";
import React from "react";
import type { IconType } from "react-icons";

type Props = {
  onClick?: (e: React.MouseEvent<HTMLElement>) => void;
  postIcon?: IconType;
  onIconClick?: (e: any) => void;
  textDecoration?: "line-through" | "underline" | "overline";
  className?: string;
  children: React.ReactNode;
};

export const ListItem: React.FC<Props> = ({
  onClick,
  textDecoration,
  postIcon: PostIcon,
  onIconClick,
  className,
  children,
}) => {
  return (
    <li
      className={clsx(
        "flex min-h-[68px] w-full items-center justify-between gap-x-4 rounded-2xl bg-gray-800 p-5",
        textDecoration,
        className
      )}
      onClick={onClick}
    >
      <span className="flex gap-x-4">{children}</span>
      {PostIcon && (
        <PostIcon
          className="h-6 w-6 shrink-0 text-gray-400 hover:text-gray-300"
          onClick={onIconClick}
        />
      )}
    </li>
  );
};
