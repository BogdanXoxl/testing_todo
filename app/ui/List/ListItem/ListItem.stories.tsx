import type { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { FiTrash2 } from "react-icons/fi";

import { Checkbox } from "../../Checkbox";
import { ListItem } from "./ListItem";

export default {
  title: "UI/ListItem",
  component: ListItem,
  argTypes: {
    className: {
      type: "string",
      control: false,
    },
    postIcon: {
      control: false,
    },
    children: {
      type: "string",
      defaultValue: "children",
    },
  },
} as ComponentMeta<typeof ListItem>;

const Template: ComponentStory<typeof ListItem> = (args) => (
  <div className="flex flex-col gap-y-4">
    <ListItem {...args} />
    <ListItem {...args} postIcon={FiTrash2} />
    <ListItem {...args}>
      <Checkbox />
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aperiam debitis deleniti eius
      impedit praesentium quas.
    </ListItem>
  </div>
);

export const Default = Template.bind({});
