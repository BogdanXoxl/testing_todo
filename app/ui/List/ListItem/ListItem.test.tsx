import { render } from "@testing-library/react";
import React from "react";
import { FiTrash2 } from "react-icons/fi";

import { Checkbox } from "../../Checkbox";
import { ListItem } from "./ListItem";

describe("Test ListItem UI component", () => {
  it("Snapshot", () => {
    const view = render(
      <ListItem postIcon={FiTrash2}>
        <Checkbox checked />
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aperiam debitis deleniti eius
        impedit praesentium quas.
      </ListItem>
    );
    expect(view).toMatchSnapshot();
  });
});
