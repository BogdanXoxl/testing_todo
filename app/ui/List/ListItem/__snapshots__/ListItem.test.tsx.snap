// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ListItem UI component Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="flex min-h-[68px] w-full items-center justify-between gap-x-4 rounded-2xl bg-neutral p-5 text-white"
      >
        <span
          class="flex gap-x-4"
        >
          <div
            class="h-7 w-7 shrink-0 rounded-lg border-2 border-primary bg-primary"
          >
            <svg
              class="text-secondary"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.97 4.97a.75.75 0 0 1 1.07 1.05l-3.99 4.99a.75.75 0 0 1-1.08.02L4.324 8.384a.75.75 0 1 1 1.06-1.06l2.094 2.093 3.473-4.425a.267.267 0 0 1 .02-.022z"
              />
            </svg>
          </div>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aperiam debitis deleniti eius impedit praesentium quas.
        </span>
        <svg
          class="h-6 w-6 shrink-0 cursor-pointer text-secondary-light-max hover:text-secondary-light-min"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="3 6 5 6 21 6"
          />
          <path
            d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
        </svg>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="flex min-h-[68px] w-full items-center justify-between gap-x-4 rounded-2xl bg-neutral p-5 text-white"
    >
      <span
        class="flex gap-x-4"
      >
        <div
          class="h-7 w-7 shrink-0 rounded-lg border-2 border-primary bg-primary"
        >
          <svg
            class="text-secondary"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.97 4.97a.75.75 0 0 1 1.07 1.05l-3.99 4.99a.75.75 0 0 1-1.08.02L4.324 8.384a.75.75 0 1 1 1.06-1.06l2.094 2.093 3.473-4.425a.267.267 0 0 1 .02-.022z"
            />
          </svg>
        </div>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aperiam debitis deleniti eius impedit praesentium quas.
      </span>
      <svg
        class="h-6 w-6 shrink-0 cursor-pointer text-secondary-light-max hover:text-secondary-light-min"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="3 6 5 6 21 6"
        />
        <path
          d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
        />
        <line
          x1="10"
          x2="10"
          y1="11"
          y2="17"
        />
        <line
          x1="14"
          x2="14"
          y1="11"
          y2="17"
        />
      </svg>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
